Content.Server/Delivery/DeliverySystem.cs

```diff
@@ -73,10 +84,68 @@ protected override void GrantSpesoReward(Entity<DeliveryComponent?> ent)
         if (!TryComp<StationBankAccountComponent>(ent.Comp.RecipientStation, out var account))
             return;
 
+        var stationAccountEnt = (ent.Comp.RecipientStation.Value, account);
+
+        var multiplier = GetDeliveryMultiplier(ent!); // Resolve so we know it's got the component
+
         _cargo.UpdateBankAccount(
-            (ent.Comp.RecipientStation.Value, account),
-            ent.Comp.SpesoReward,
-            _cargo.CreateAccountDistribution((ent.Comp.RecipientStation.Value, account)));
+            stationAccountEnt,
+            (int)(ent.Comp.BaseSpesoReward * multiplier),
+           _cargo.CreateAccountDistribution((ent.Comp.RecipientStation.Value, account)));
+    }
+
+    /// <summary>
+    /// Runs the penalty logic: Announcing the penalty and calculating how much to charge the designated account
+    /// </summary>
+    /// <param name="ent">The delivery for which to run the penalty.</param>
+    /// <param name="reasonLoc">The penalty reason, displayed in front of the message.</param>
+    protected override void HandlePenalty(Entity<DeliveryComponent> ent, string? reasonLoc = null)
+    {
+        if (!TryComp<StationBankAccountComponent>(ent.Comp.RecipientStation, out var stationAccount))
+            return;
+
+        if (ent.Comp.WasPenalized)
+            return;
+
+        var multiplier = GetDeliveryMultiplier(ent);
+
+        var accountName = _protoMan.TryIndex(ent.Comp.PenaltyBankAccount, out var accountInfo) &&
+                   Loc.TryGetString(accountInfo.Name, out var localizedAccountName)
+                  ? localizedAccountName
+                  : Loc.GetString(UnknownAccount);
+
+        if (reasonLoc == null || !Loc.TryGetString(reasonLoc, out var reason))
+            reason = Loc.GetString(DefaultMessage);
+
+        var dist = new Dictionary<ProtoId<CargoAccountPrototype>, double>()
+        {
+            { ent.Comp.PenaltyBankAccount, 1.0 }
+        };
+
+        var penaltyAccountBalance = stationAccount.Accounts[ent.Comp.PenaltyBankAccount];
+        var calculatedPenalty = (int)(ent.Comp.BaseSpesoPenalty * multiplier);
+
+        // Prevents cargo from going into negatives
+        if (penaltyAccountBalance - calculatedPenalty < 0)
+            calculatedPenalty = Math.Max(0, penaltyAccountBalance);
```

Comment:
```suggestion
        calculatedPenalty = Math.Max(0, penaltyAccountBalance);
```
that is already doing the comparison

Reply:
nevermind, this is different